sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -269,3 +269,62 @@ case class TypeOf(child: Expression) extends UnaryExpression {
     defineCodeGen(ctx, ev, _ => s"""UTF8String.fromString(${child.dataType.catalogString})""")
   }
 }
+
+@ExpressionDescription(
+  usage = """_FUNC_(expr) - Execute all children and return the last child result.""",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(1, 2);
+       2
+      > SELECT _FUNC_(1 + 2, 3 + 4);
+       7
+  """,
+  since = "3.2.0",
+  group = "misc_funcs")
+case class DelegateFunction(children: Seq[Expression]) extends Expression {
+  require(children.nonEmpty, s"$prettyName function requires children is not empty.")
+
+  private lazy val lastChild = children.last
+
+  override lazy val deterministic: Boolean = children.forall(_.deterministic)
+  override lazy val resolved: Boolean = children.forall(_.resolved)
+  override def foldable: Boolean = children.forall(_.foldable)
+  override def nullable: Boolean = lastChild.nullable
+  override def dataType: DataType = lastChild.dataType
+
+  override def eval(input: InternalRow): Any = {
+    var result: Any = null
+    children.foreach { child =>
+      result = child.eval(input)
+    }
+    result
```

Comment:
Hmmm how about add a result map and avoid re-compute same child?

Reply:
Not sure what do you mean `same child` ?  This function just execute child one by one.